NDTensors/ext/NDTensorsMetalExt/permutedims.jl

```diff
@@ -1,3 +1,10 @@
+function Base.permutedims!(
+  Edest::Exposed{<:MtlArray,<:Base.ReshapedArray}, Esrc::Exposed{<:MtlArray}, perm
+)
+  Eperm = expose(permutedims(Esrc, perm))
```

Comment:
The way you have it written now, `expose` isn't doing anything since you just unwrap it in the next line.

Perhaps you want:
```julia
function Base.permutedims!(
  Edest::Exposed{<:MtlArray,<:Base.ReshapedArray}, Esrc::Exposed{<:MtlArray}, perm
)
  Eperm = expose(permutedims(Esrc, perm))
  copyto!(expose(parent(Edest.object)), Eperm)
  return Edest.object
end
```
Note that here I'm trying to ensure `Exposed` objects get passed to `copyto!`, in case there are issues that out own version of `copyto!` needs to fix for `MtlArray`.

I think there are a few things we need to establish in terms of conventions and semantics of `expose`:
1. We should define `unexpose(e::Exposed) = e.object` and prefer to use `unexpose`.
2. What should out-of-place functions that take `Exposed` objects as inputs return? I think they should output an unexposed object, i.e. `identity(e::Exposed) = unexpose(e)`, `permutedims(e::Exposed, perm) = permutedims(unexposed(e), perm)`, etc.
3. What should in-place functions that take `Exposed` objects return? I think they should return the modified unexposed object, i.e. `permutedims!(e::Exposed, perm) = (permutedims!(unexposed(e), perm); return unexposed(e))`.
4. Like I show above, we need to think carefully about when we `expose` or `unexpose` objects to not overuse it but try to anticipate potential issues that `expose` can fix.